### PR TITLE
OkHttp: sync error map with gRpc spec.

### PR DESCRIPTION
**Pull request by madongfly**
See [grpc spec](https://github.com/grpc/grpc-common/blob/master/PROTOCOL-HTTP2.md#errors)

@ejona86, please take a look, thanks!

### CODE REVIEW

File: okhttp/src/main/java/io/grpc/transport/okhttp/OkHttpClientTransport.java [hunk 1]
```diff
@@ -116,12 +109,17 @@
...
     errorToStatus.put(ErrorCode.COMPRESSION_ERROR,
         Status.INTERNAL.withDescription("Compression error"));
-    errorToStatus.put(ErrorCode.INVALID_CREDENTIALS,
-        Status.PERMISSION_DENIED.withDescription("Invalid credentials"));
+    errorToStatus.put(ErrorCode.CONNECT_ERROR,
+        Status.INTERNAL.withDescription("Connect error"));
+    errorToStatus.put(ErrorCode.ENHANCE_YOUR_CALM,
+        Status.RESOURCE_EXHAUSTED.withDescription("Enhance your calm"));
+    errorToStatus.put(ErrorCode.INADEQUATE_SECURITY,
+        Status.PERMISSION_DENIED.withDescription("Enhance your calm"));
```
Comment (ejona86) [inline okhttp/src/main/java/io/grpc/transport/okhttp/OkHttpClientTransport.java]: Fix description.

Comment (madongfly) [inline okhttp/src/main/java/io/grpc/transport/okhttp/OkHttpClientTransport.java]: Thanks for the catch, done.

**Comment by madongfly** [comment]
@ejona86 PTAL, thanks!

**Comment by ejona86** [comment]
@madongfly LGTM

**Pull request merged**